shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -477,6 +488,72 @@ public boolean performEditorAction(int actionCode) {
     return true;
   }
 
+  @Override
+  @TargetApi(25)
+  @RequiresApi(25)
+  public boolean commitContent(InputContentInfo inputContentInfo, int flags, Bundle opts) {
+    // Ensure permission is granted.
+    if (Build.VERSION.SDK_INT >= 25
+        && (flags & InputConnectionCompat.INPUT_CONTENT_GRANT_READ_URI_PERMISSION) != 0) {
+      try {
+        inputContentInfo.requestPermission();
+      } catch (Exception e) {
+        return false;
+      }
+    } else {
+      return false;
+    }
+
+    if (inputContentInfo.getDescription().getMimeTypeCount() > 0) {
+      inputContentInfo.requestPermission();
+
+      final Uri uri = inputContentInfo.getContentUri();
+      final String mimeType = inputContentInfo.getDescription().getMimeType(0);
+      Context context = mFlutterView.getContext();
+
+      if (uri != null) {
+        InputStream is;
+        try {
+          // Extract byte data from the given URI.
+          is = context.getContentResolver().openInputStream(uri);
+        } catch (FileNotFoundException ex) {
+          return false;
+        }
+
+        if (is != null) {
+          final byte[] data = this.readStreamFully(is, 64 * 1024);
+
+          final Map<String, Object> obj = new HashMap<>();
+          obj.put("mimeType", mimeType);
+          obj.put("data", data);
+          obj.put("uri", uri.toString());
+
+          // Commit the content to the text input channel and release the permission.
+          textInputChannel.commitContent(mClient, obj);
+          inputContentInfo.releasePermission();
```

Comment:
Do you have to release permission when `is` is null? You request it in line 508 but only release when successful here.

Reply:
Fixed. Good catch :)